torch/csrc/jit/export.cpp

```diff
@@ -577,6 +577,11 @@ class ScriptModuleSerializer {
     std::stringstream fname;
     fname << archive_name << ".pkl";
     writer_.writeRecord(fname.str(), data.data(), data.size());
+    // write all the captured run-time class types during pickling the IValues
```

Comment:
nit: This setup is slightly different than the way the python printer works. The python printer takes a reference to a vector and appends the class dependencies. Can we make the pickler do the same? When I land https://github.com/pytorch/pytorch/pull/26788 I am going to have reconcile to two and it would be better if they both could write to the class_deps_ array.